task.sql

```diff
@@ -1,33 +1,56 @@
--- Create database and tables
-
+-- Drop and recreate the database
+DROP DATABASE IF EXISTS ShopDB;
 CREATE DATABASE ShopDB;
 USE ShopDB;
 
+-- Create Countries table
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 );
 
+-- Create Products table
+CREATE TABLE Products (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    PRIMARY KEY (ID)
+);
+
+-- Create Warehouses table
+CREATE TABLE Warehouses (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    Address VARCHAR(100),
+    CountryID INT,
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION
+);
+
+-- Create ProductInventory table
 CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
+    ID INT AUTO_INCREMENT,
+    ProductID INT,
+    WarehouseID INT,
     WarehouseAmount INT,
-    WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
-    CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+    PRIMARY KEY (ID),
+    FOREIGN KEY (ProductID) REFERENCES Products(ID) ON DELETE CASCADE,
+    FOREIGN KEY (WarehouseID) REFERENCES Warehouses(ID) ON DELETE CASCADE
```

Comment:
The foreign key constraints in the 'ProductInventory' table use 'ON DELETE CASCADE'. Ensure this behavior is intended, as it will delete inventory records if the corresponding product or warehouse is deleted.